lib/puppet/util/monkey_patches.rb

```diff
@@ -20,6 +20,13 @@ def self.caller(skip=nil)
 
 class Symbol
   def <=> (other)
+    if (other.class != Symbol)
+      if Puppet[:strict] == :warning
+        Puppet.warn_once('deprecation', 'symbol_comparison', 'Comparing Symbols to non-Symbol values is deprecated')
+      elsif Puppet[:strict] == :error
+        raise ArgumentError.new("Comparing Symbols to non-Symbol values is no longer allowed")
```

Comment:
You're right. I've been spoiled by optimizing compilers (and much less flexible case statements) off in C++ land ;)

I'll clean that up